### PR TITLE
Restore chart view

### PR DESCRIPTION
**Pull request by morgandruesne**
## What does this PR do ?
This PR restores the chart view using vue-apexcharts

### CODE REVIEW

**Review by jenow — APPROVED** [review]

File: src/components/Data/Documents/Views/TimeSeries.vue [hunk 1]
```diff
@@ -213,25 +207,25 @@ export default {
...
-      this.chart.push(chartTitles)
-
-      for (const doc of this.documents) {
-        const item = []
-        item.push(doc.content[this.customDateField])
-        for (const field of this.customNumberFields) {
-          item.push(_.get(doc.content, field.name, ''))
+      let series = []
+      for (const field of this.customNumberFields) {
+        let data = {
```
Comment (Aschen) [inline src/components/Data/Documents/Views/TimeSeries.vue:217]:
```suggestion
        const data = {
```

File: src/components/Data/Documents/Views/TimeSeries.vue [hunk 2]
```diff
@@ -213,25 +207,25 @@ export default {
...
-        chartTitles.push(item.name)
       }
-      this.chart.push(chartTitles)
-
-      for (const doc of this.documents) {
-        const item = []
-        item.push(doc.content[this.customDateField])
-        for (const field of this.customNumberFields) {
-          item.push(_.get(doc.content, field.name, ''))
+      let series = []
```
Comment (Aschen) [inline src/components/Data/Documents/Views/TimeSeries.vue:215]:
```suggestion
      const series = []
```

File: src/components/Data/Documents/Views/TimeSeries.vue (hunk 1 above)
Comment (Aschen) [reply src/components/Data/Documents/Views/TimeSeries.vue:217]: Also having `data.data` is a bit confusing :sweat_smile: 
IMHO `serie` is a better name for this variable

File: src/components/Data/Documents/Views/TimeSeries.vue [hunk 3]
```diff
@@ -213,25 +207,25 @@ export default {
...
+          data: []
+        }
+        for (const doc of this.documents) {
+          data.data.push(doc[field.name])
         }
-        this.chart.push(item)
+        series.push(data)
       }
+      if (this.$refs.Chart)
+        this.$refs.Chart.updateOptions(this.chartOptions)
```
Comment (Aschen) [inline src/components/Data/Documents/Views/TimeSeries.vue:227]:
```suggestion
      if (this.$refs.Chart) {
        this.$refs.Chart.updateOptions(this.chartOptions)
      }
```

File: src/components/Data/Documents/Views/TimeSeries.vue [hunk 4]
```diff
@@ -1,28 +1,26 @@
...
+      <div class="col s12">
         <span>Date</span>
         <autocomplete
           placeholder="Date field"
           :items="mappingDateArray"
           :value="customDateField || ''"
           :notify-change="false"
           @autocomplete::change="
-            item => {
+            (item) => {
```
Comment (Aschen) [inline src/components/Data/Documents/Views/TimeSeries.vue:23]: Nitpicking: why did you add the parenthesis? We only need them with at least 2 variables

File: src/components/Data/Documents/Views/TimeSeries.vue [hunk 5]
```diff
@@ -47,7 +45,7 @@
             :new-value="newCustomNumberField || ''"
             @update-color="updateColor"
             @autocomplete::change="
-              item => {
+              (item) => {
```
Comment (Aschen) [inline src/components/Data/Documents/Views/TimeSeries.vue:48]: Nitpicking: why did you add the parenthesis? We only need them with at least 2 variables

**Review by Aschen — CHANGES_REQUESTED** [review]
Good to see you again @morgandruesne :+1:

**Review by alexandrebouthinon — APPROVED** [review]

File: src/components/Data/Collections/DropdownView.vue [hunk 6]
```diff
@@ -25,6 +25,13 @@
         >
           Column view
         </b-dropdown-item>
+        <b-dropdown-item
+          data-cy="CollectionDropdown-TimeSeries"
+          :active="activeView === 'time-series'"
+          @click="$emit('time-series')"
+        >
+          chart view
```
Comment (berthieresteban) [inline src/components/Data/Collections/DropdownView.vue:33]:
```suggestion
          Chart view
```

**Review by berthieresteban — CHANGES_REQUESTED** [review]

**Comment by berthieresteban** [comment]
Nice !!! 
Some errors might not be handled, I used to try the chart view but nothing was displayed and I a had an console error from apexchart:
image

File: src/components/Data/Documents/Views/TimeSeries.vue (hunk 5 above)
Comment (morgandruesne) [reply src/components/Data/Documents/Views/TimeSeries.vue]: I think it's an extension of my editor, I'll see if I can fix it.

File: src/components/Data/Documents/Views/TimeSeries.vue [hunk 7]
```diff
@@ -213,25 +208,26 @@ export default {
...
-      this.chart.push(chartTitles)
-
-      for (const doc of this.documents) {
-        const item = []
-        item.push(doc.content[this.customDateField])
-        for (const field of this.customNumberFields) {
-          item.push(_.get(doc.content, field.name, ''))
+      const series = []
+      for (const field of this.customNumberFields) {
+        const data = {
```
Comment (Aschen) [inline src/components/Data/Documents/Views/TimeSeries.vue:218]:
```suggestion
        const serie = {
```

File: src/components/Data/Documents/Views/TimeSeries.vue [hunk 8]
```diff
@@ -213,25 +208,26 @@ export default {
...
+          name: field.name,
+          data: []
         }
-        this.chart.push(item)
+        for (const doc of this.documents) {
+          data.data.push(_.get(doc, field.name, ''))
```
Comment (Aschen) [inline src/components/Data/Documents/Views/TimeSeries.vue:223]:
```suggestion
          serie.data.push(_.get(doc, field.name, ''))
```

File: src/components/Data/Documents/Views/TimeSeries.vue [hunk 9]
```diff
@@ -213,25 +208,26 @@ export default {
...
+        }
+        series.push(data)
```
Comment (Aschen) [inline src/components/Data/Documents/Views/TimeSeries.vue:225]:
```suggestion
        series.push(serie)
```

**Review by Aschen — APPROVED** [review]

**Comment by berthieresteban** [comment]
Still having a problem, but it's better like that, not having console errors nor displat bugs !
But I cant select any fields to display the chart there is just a `no data to display` message but the pagination still displayed ..
image

**Comment by berthieresteban** [comment]
Great job ! It's almost over !!!

Just 2 last things to be iso v3:
  - We need to implement the itemsPerPage option (maybe you could copy it from list vue)
  - The labels on x axis aren't displaying any interesting data and as they wasn't there in v3 we probably should remove them here

**Review by Njuelle — CHANGES_REQUESTED** [review]
we really need to write some tests for this feature

**Review by xbill82 — APPROVED** [review]

**Comment by berthieresteban** [comment]
Here is a sh script to create fixtures to test this pr:
```
#!/bin/bash
kourou index:delete testindex
kourou index:create testindex
kourou collection:create testindex testchart '{"mappings": {"properties": {"temperature": {"type": "integer"}, "battery": {"type": "integer"}, "payloadDate": {"type": "date"}}}}'

documents="{documents:["
for i in {1..50}
do
  now=$(date +%s%N | cut -b1-13)
  date=$(($now - $i * 3600000))
  temp=$(shuf -i 10-40 -n 1)
  battery=$(shuf -i 0-100 -n 1)
  documents="$documents {body: {temperature: $temp, battery: $battery, payloadDate: $date}},"
done
documents="${documents::-1}]}"

kourou document:mCreate testindex testchart "$documents"

**Comment by xbill82** [comment]
Converted to draft while we fix some timestamp problems

**Review by berthieresteban — APPROVED** [review]

File: src/components/Data/Documents/Views/TimeSeries.vue [hunk 10]
```diff
@@ -53,16 +52,55 @@
...
 </template>
 
 <script>
-import Autocomplete from '../../../Common/Autocomplete'
 import TimeSeriesItem from './TimeSeriesItem'
-import { GChart } from 'vue-google-charts'
+import VueApexCharts from 'vue-apexcharts'
 import _ from 'lodash'
+import { dateFromTimestamp } from '@/utils'
+// import moment from 'moment'
```
Comment (Njuelle) [inline src/components/Data/Documents/Views/TimeSeries.vue:103]:
```suggestion
```

**Review by Njuelle — APPROVED** [review]
